jetty-util/src/main/java/org/eclipse/jetty/util/ssl/SslKeyStoreScanner.java

```diff
@@ -0,0 +1,145 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2020 Mort Bay Consulting Pty Ltd and others.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.util.ssl;
+
+import java.io.File;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.jetty.util.Scanner;
+import org.eclipse.jetty.util.annotation.ManagedAttribute;
+import org.eclipse.jetty.util.annotation.ManagedOperation;
+import org.eclipse.jetty.util.component.AbstractLifeCycle;
+import org.eclipse.jetty.util.log.Log;
+import org.eclipse.jetty.util.log.Logger;
+
+public class SslKeyStoreScanner extends AbstractLifeCycle implements Scanner.DiscreteListener
+{
+    private static final Logger LOG = Log.getLogger(SslKeyStoreScanner.class);
+
+    private final SslContextFactory sslContextFactory;
+    private final File keystoreFile;
+    private final List<File> files = new ArrayList<>();
+    private Scanner _scanner;
+    private int _scanInterval = 1;
+
+    public SslKeyStoreScanner(SslContextFactory sslContextFactory)
+    {
+        this.sslContextFactory = sslContextFactory;
+        this.keystoreFile = new File(URI.create(sslContextFactory.getKeyStorePath())); // getKeyStorePath is giving url instead of path?
+        if (!keystoreFile.exists())
+            throw new IllegalArgumentException("keystore file does not exist");
+        if (keystoreFile.isDirectory())
+            throw new IllegalArgumentException("expected keystore file not directory");
+
+        File parentFile = keystoreFile.getParentFile();
+        if (!parentFile.exists() || !parentFile.isDirectory())
+            throw new IllegalArgumentException("error obtaining keystore dir");
+
+        files.add(parentFile);
+        if (LOG.isDebugEnabled())
+            LOG.debug("created {}", this);
+    }
+
+    @Override
+    protected void doStart() throws Exception
+    {
+        if (LOG.isDebugEnabled())
+            LOG.debug(this.getClass().getSimpleName() + ".doStart()");
+
+        _scanner = new Scanner();
```

Comment:
Make `_scanner` `final` and add it as a bean in the constructor.
Here just configure it and call `super` to start it.